client/verta/tests/unit_tests/test_deployed_model.py

```diff
@@ -0,0 +1,377 @@
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict_with_id(x=['test_prediction'])
+
+
+def test_predict_with_id_altered_retry_config(mocked_responses) -> None:
+    """ Calling predict with custom retry parameters changes the retry config and makes the correct requests. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=888,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict_with_id(
+            x=['test_prediction'],
+            max_retries=9,
+            retry_status={888},
+            backoff_factor=0.1
+            )
+        mocked_responses.assert_call_count(PREDICTION_URL, 10)
+
+
+def test_default_retry_after_custom_retry(mocked_responses) -> None:
+    """ Calling predict with default params after calling predict with custom
+        params uses default retry settings and not the custom settings from
+        the previous call. """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=777,
+        )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(
+            x=['test_prediction'],
+            max_retries=1,
+            retry_status={777},
+            backoff_factor=0.1,
+            )
+        mocked_responses.assert_call_count(PREDICTION_URL, 2)
+        # 1 attempt + 1 retry = 2
+
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={},
+        status=429,
+        )
+    with pytest.raises(RetryError):
+        dm.predict(x=['test_prediction'])  # use defaults
+        mocked_responses.assert_call_count(PREDICTION_URL, 16)
+        # previous 2 + 1 attempt + default 13 retries = 16
+
+
+def test_predict_400_error_message_extraction(mocked_responses) -> None:
+    """ Getting a 400 will render the attached message form the backend if present """
+    mocked_responses.post(
+        PREDICTION_URL,
+        json={"message": "Here be a message in the response"},
+        status=400,
+        headers={'verta-request-id': 'hereISthisTESTidFROMtheUSER'},
+    )
+    creds = EmailCredentials.load_from_os_env()
+    dm = DeployedModel(
+        prediction_url=PREDICTION_URL,
+        creds=creds,
+        token=TOKEN,
+    )
+    with pytest.raises(RuntimeError) as err:
+        dm.predict(x=['test_prediction'])
+    assert str(err.value) == (
+        'Deployed model encountered an error: Here be a message in the response'
+        )
+
+
+def test_predict_400_error_message_missing(mocked_responses) -> None:
+    """ Getting a 400 will render the attached message form the backend if present """
```

Comment:
The name and description of this test don't match what it's doing.

Reply:
@ewagner-verta This still doesn't seem correct to me. The test name and docstring say `400` but the test is returning and handling a `500`